config/initializers/rpush.rb

```diff
@@ -0,0 +1,144 @@
+Rpush.configure do |config|
+  # Supported clients are :active_record and :redis
+  config.client = :redis
+
+  # Options passed to Redis.new
+  # config.redis_options = {}
+
+  # Frequency in seconds to check for new notifications.
+  config.push_poll = 2
+
+  # The maximum number of notifications to load from the store every `push_poll` seconds.
+  # If some notifications are still enqueued internally, Rpush will load the batch_size less
+  # the number enqueued. An exception to this is if the service is able to receive multiple
+  # notification payloads over the connection with a single write, such as APNs.
+  config.batch_size = 100
+
+  # Path to write PID file. Relative to current directory unless absolute.
+  config.pid_file = "tmp/rpush.pid"
+
+  # Path to log file. Relative to current directory unless absolute.
+  config.log_file = "log/rpush.log"
+
+  config.log_level = (defined?(Rails) && Rails.logger ? Rails.logger.level : ::Logger::Severity::INFO)
+
+  # Define a custom logger.
+  # config.logger = MyLogger.new
+
+  # By default in foreground mode logs are directed both to the logger and to stdout.
+  # If the logger goes to stdout, you can disable foreground logging to avoid duplication.
+  # config.foreground_logging = false
+
+  # config.apns.feedback_receiver.enabled = true
+  # config.apns.feedback_receiver.frequency = 60
+end
+
+Rpush.reflect do |on|
+  # Called with a Rpush::Apns::Feedback instance when feedback is received
+  # from the APNs that a notification has failed to be delivered.
+  # Further notifications should not be sent to the device.
+  # on.apns_feedback do |feedback|
+  # end
+
+  # Called when a notification is queued internally for delivery.
+  # The internal queue for each app runner can be inspected:
+  #
+  # Rpush::Daemon::AppRunner.runners.each do |app_id, runner|
+  #   runner.app
+  #   runner.queue_size
+  # end
+  #
+  # on.notification_enqueued do |notification|
+  # end
+
+  # Called when a notification is successfully delivered.
+  # on.notification_delivered do |notification|
+  # end
+
+  # Called when notification delivery failed.
+  # Call 'error_code' and 'error_description' on the notification for the cause.
+  on.notification_failed do |notification|
+    if notification.error_description == "BadDeviceToken"
+      # When adding new platforms we'll need to check to only clear out devices
+      # scoped to that specific platform. i.e. notification.class.to_s =~ /Apns.+::Notification/
+      Device.where(token: notification.device_token, platform: "iOS").destroy_all
+    end
+
+    DatadogStatsClient.increment("push_notifications.errors",
```

Comment:
We shouldn't use `DatadogStatsClient` directly anymore, see https://github.com/forem/forem/pull/12304

```suggestion
    ForemStatsClient.increment("push_notifications.errors",
```

Reply:
@joshpuetz Highlighting this suggestion again, I think we should standardize around the new stats client.

Reply:
Whoops, totally missed this the first time around, thanks for calling it out again!